### PR TITLE
refactor: crunchy adapted to use SocketAddr instead of IpAddr

### PR DESCRIPTION
**Pull request by asmie**
Refactoring PR incorporating usage of SocketAddr instead of IpAddr in whole project to ensure that each node or peer is fully identified with IP address and port number to distinguish possible different instances withing single IP address.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,22 +33,21 @@ The generated output contains processed data that our renderer can directly use.
...
         connections: Vec<usize>,
-        geolocation: GeoInfo
+        geolocation: Option<GeoInfo>
     ]
 }
 ```
 Explaination of the node fields:
 
-- `ip`: the address as dotted quad, without port number
+- `addr`: the address as dotted quad, wit port number
```
Comment (Rqnsom) [inline README.md:50]:
```suggestion
- `addr`: the address as a dotted quad, with port number
```

**Review by Rqnsom — APPROVED** [review]
LGTM

**Review by kylegranger — APPROVED** [review]
LGTM! 👍  🚀

File: README.md (hunk 1 above)
Comment (asmie) [reply README.md]: Corrected.

**Pull request merged**